packages/aws-cdk/toolkit/lib/cli-io-host.ts

```diff
@@ -0,0 +1,101 @@
+import * as chalk from 'chalk';
+
+/**
+ * Basic message structure for toolkit notifications.
+ * Messages are emitted by the toolkit and handled by the IoHost.
+ */
+interface IoMessage {
+  time: Date;
+  level: IoMessageLevel;
+  action: IoAction;
+  code: string;
+  message: string;
+  forceStdout?: boolean;
+  // Optionally specify Chalk style for stdout/stderr, if TTY is enabled
+  style?: ((str: string) => string);
```

Comment:
[reviving the discussion on styling]

> I thought this would be a nice thing to add for customers when we create the IoHost interface

i'm not opinionated about this either way, but isn't the point of `CliIoHost` to be an unopinionated 1-1 matching to what we currently have in the CDK CLI? I don't see us deciding to use custom styling beyond what our current logging styles look like.

Reply:
I personally want this to be part of the `IoMessage` interface when we define the base `IoHost` interface. However thinking about it now I think we can be clever here and allow customers to style their messages with chalk however they want but that'll be apart of my chalk support pr

Reply:
Gonna delete this for now